File: flight-tracking-service/src/main/java/github/wozniak/flighttrackingservice/core/dto/AircraftDTO.java
package github.wozniak.flighttrackingservice.core.dto;

import github.wozniak.flighttrackingservice.core.entity.Aircraft;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class AircraftDTO {
    private String make;
    private String model;
    private String type;
    private Integer seatingCapacity;
    private Integer cruisingSpeedKnots;
    private Integer rangeMiles;

    public AircraftDTO(Aircraft model){
        this.make = model.getMake();
        this.model = model.getModel();
        this.type = model.getAircraftCategory().toString();
        this.seatingCapacity = model.getSeatingCapacity();
        this.cruisingSpeedKnots = model.getCruisingSpeedKnots();
        this.rangeMiles = model.getRangeMiles();
    }

    public static List<AircraftDTO> fromList(List<Aircraft> models){
        return models.stream().map(AircraftDTO::new).toList();
    }
}

File: backend/src/Domain/aircraft/Dto/AircraftDto.cs
using backend.Domain.aircraft.Entity;
using backend.Domain.aircraft.Enums;

namespace backend.Domain.aircraft.Dto;

public class AircraftDto
{
    public string Manufacturer { get; set; }
    public string Model { get; set; }
    public int CruiseSpeedKnots { get; set; }
    public int RangeNauticalMiles { get; set; }
    public int PassengerCapacity { get; set; }
    public string Category { get; set; }

    public AircraftDto(Aircraft aircraft)
    {
        Manufacturer = aircraft.Manufacturer.ToName();
        Model = aircraft.Model;
        CruiseSpeedKnots = aircraft.CruiseSpeedKnots;
        RangeNauticalMiles = aircraft.RangeNauticalMiles;
        PassengerCapacity = aircraft.PassengerCapacity;
        Category = aircraft.Category.ToString();
    }
}